etl/steps/grapher/bp/2022-07-14/statistical_review.py

```diff
@@ -12,16 +10,10 @@
 DATASET_PATH = DATA_DIR / "garden" / "bp" / "2022-07-14" / "statistical_review"
 
 
-def get_grapher_dataset() -> catalog.Dataset:
-    dataset = catalog.Dataset(DATASET_PATH)
-    dataset.metadata = gh.adapt_dataset_metadata_for_grapher(dataset.metadata)
-
-    return dataset
+def run(dest_dir: str) -> None:
+    garden_dataset = catalog.Dataset(DATASET_PATH)
+    dataset = catalog.Dataset.create_empty(dest_dir, gh.adapt_dataset_metadata_for_grapher(garden_dataset.metadata))
 
-
-def get_grapher_tables(dataset: catalog.Dataset) -> Iterable[catalog.Table]:
     # There is only one table in the dataset, with the same name as the dataset.
-    table = dataset[dataset.table_names[0]].reset_index().drop(columns=["country_code"])
-    table = gh.adapt_table_for_grapher(table)
-
-    yield from gh.yield_wide_table(table, na_action="drop")
+    table = garden_dataset[garden_dataset.table_names[0]].reset_index().drop(columns=["country_code"])
+    dataset.add(gh.adapt_table_for_grapher(table))
```

Comment:
Missing `dataset.save()` at the end, to update table metadata?

Perhaps we should add a `__del__()` method to `Dataset` that does this automatically, since it's not obvious that it's expected or needed.

Reply:
I saw the `.save` being called right after creating a dataset in one of the very first examples in ETL (I think it was hyde) and it sticked with me. I thought it has a special reason, but it doesn't. I also find calling `save` at the end more intuitive.

Since this is being used in pretty much all recipes we have, can I create a separate issue & PR for it? Table metadata is updated correctly right now (unless someone changes dataset metadata after calling save, but that's not happening).

Reply:
Yeah, a separate PR is a good idea 👍